og_migrate/src/Plugin/migrate/OgEntityDeriver.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+namespace Drupal\og_migrate\Plugin\migrate;
+
+use Drupal\Component\Plugin\ConfigurableInterface;
```

Comment:
This is the cause of failure on 8.6.x because it was introduced in 8.7.x and the previous class has been deprecated and will be removed completely in 9.0.0.

I've been developing on 8.7.x (or whatever the latest core dev branch is) as part of my local development work flow.

The testing matrix should be updated now that 8.7.x is in alpha.

Reply:
Thanks @mradcliffe for your updates!

We're already testing 8.7.x:

```
php:
  - 7.1
  - 7.2
  - 7.3

env:
  global:
    - COMPOSER_MEMORY_LIMIT=2G
  matrix:
    - TEST_SUITE=8.6.x
    - TEST_SUITE=8.7.x
    - TEST_SUITE=PHP_CodeSniffer

# Only run the coding standards check once.
matrix:
  exclude:
    - php: 7.1
      env: TEST_SUITE=PHP_CodeSniffer
    - php: 7.2
      env: TEST_SUITE=PHP_CodeSniffer
  allow_failures:
    - env: TEST_SUITE=8.7.x

```
from https://github.com/Gizra/og/blob/8.x-1.x/.travis.yml

Note that on 8.7.x there are some errors too:
```
1) Drupal\Tests\og\Kernel\Access\OgGroupContentOperationAccessTest::testAccess with data set #0 ('newsletter', array(array(array(true), array(true, true), array(true, true)), array(array(true), array(true, true), array(true, true)), array(array(true), array(true, false), array(true, false)), array(array(true), array(true, false), array(true, false)), array(array(false), array(false, false), array(false, false))))
Drupal\Core\Entity\EntityStorageException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'field_name' cannot be null: INSERT INTO {comment_field_data} (cid, comment_type, langcode, status, uid, pid, entity_id, subject, name, mail, homepage, hostname, created, changed, thread, entity_type, field_name, default_langcode) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4, :db_insert_placeholder_5, :db_insert_placeholder_6, :db_insert_placeholder_7, :db_insert_placeholder_8, :db_insert_placeholder_9, :db_insert_placeholder_10, :db_insert_placeholder_11, :db_insert_placeholder_12, :db_insert_placeholder_13, :db_insert_placeholder_14, :db_insert_placeholder_15, :db_insert_placeholder_16, :db_insert_placeholder_17); Array
(
    [:db_insert_placeholder_0] => 1
    [:db_insert_placeholder_1] => newsletter
    [:db_insert_placeholder_2] => en
    [:db_insert_placeholder_3] => 0
    [:db_insert_placeholder_4] => 1
    [:db_insert_placeholder_5] => 
    [:db_insert_placeholder_6] => 1
    [:db_insert_placeholder_7] => subscribe
    [:db_insert_placeholder_8] => 
    [:db_insert_placeholder_9] => 
    [:db_insert_placeholder_10] => 
    [:db_insert_placeholder_11] => 
    [:db_insert_placeholder_12] => 1553955275
    [:db_insert_placeholder_13] => 1553955275
    [:db_insert_placeholder_14] => 01/
    [:db_insert_placeholder_15] => entity_test
    [:db_insert_placeholder_16] => 
    [:db_insert_placeholder_17] => 1
)
/home/travis/build/Gizra/drupal/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php:847
/home/travis/build/Gizra/drupal/core/lib/Drupal/Core/Entity/EntityBase.php:394
/home/travis/build/Gizra/og/tests/src/Kernel/Access/OgGroupContentOperationAccessTest.php:245
```